### ISSUE TITLE
SBFspotUploadDaemon troubleshooting

### ISSUE THREAD

**Issue by SBFspot** [SBFspot/SBFspot#105]
Today I got an alert from PVOutput telling me my account had been idle for more than 60 mins.
My Raspi was still online and the timestamp on the SQLite database was current, so I suspected an issue with the upload daemon.  Nothing awry was showing in the logs.  My attempt to stop the daemon (killall) and restart it failed.  I'm sure a reboot will fix this by clearing /tmp, but how can I work out what has gone wrong to avoid future issues?

SBFspot version V3.0.3 running on Raspberry Pi.

pi@RASPI2 ~/smadata/logs $ uname -a
Linux RASPI2 3.12.35+ #730 PREEMPT Fri Dec 19 18:31:24 GMT 2014 armv6l GNU/Linux

pi@RASPI2 /tmp $ uptime
 20:06:01 up 14 days, 23:02,  1 user,  load average: 0.09, 0.15, 0.20
 
/home/pi/smadata/logs/SBFspotUpload20150216.log
[13:21:29] INFO: Uploading datapoint: 20150216,13:15,6398639,12,,,66.8,0 => OK (200)
[20:01:27] INFO: Stopping Daemon...
[20:02:12] INFO: Starting SBFspotUploadDeamon Version 1.0.1
[20:02:12] ERROR: Unable to open output file. See syslog for details.

/var/log/syslog
Feb 16 19:50:01 RASPI2 /USR/SBIN/CRON[25644]: (pi) CMD (/usr/local/bin/sbfspot.3/SBFspot -v -finq -nocsv >/dev/null)
Feb 16 19:55:01 RASPI2 /USR/SBIN/CRON[25649]: (pi) CMD (/usr/local/bin/sbfspot.3/SBFspot -v -finq -nocsv >/dev/null)
Feb 16 20:00:01 RASPI2 /USR/SBIN/CRON[25675]: (pi) CMD (/usr/local/bin/sbfspot.3/SBFspot -v -finq -nocsv >/dev/null)
Feb 16 20:01:27 RASPI2 SBFspotUploadDaemon: Exit signal received.
Feb 16 20:02:12 RASPI2 SBFspotUploadDaemon: Unable to open output file /tmp/SBFspotUploadDaemon.out: Success

pi@RASPI2 /tmp $ ls -lrt
total 0
-rw-r--r-- 1 root root 0 Feb  1 19:17 SBFspotUploadDaemon.out

#### Attachments
[board.jpg](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=100&FileAttachmentId=1430836)

#### This work item was migrated from CodePlex
CodePlex work item ID: '100'
Vote count: '1'

**Comment by SBFspot** [comment]
[SBF@17/02/2015]
IMO you can't rely on the datestamp of your SQLite db
To be sure have a look at the SBFspot.log if it has been enabled or examine the db for unsent records
``` SQL
SELECT * FROM vwdayData WHERE PVoutput IS NULL LIMIT 10;
```

**Comment by SBFspot** [comment]
[hardisco@17/02/2015]
Ah of course.  Excellent advice.  SQLite db had no unsent entries.  I added -v5 level logging which identified that my inverter was not responding on bluetooth.  See attached for root cause, a melted electrical circuit!!!  I've called my installer to investigate.

SBFSpot/PVOutput alerted me to this problem within an hour of it occurring.  Without these excellent services I may not have noticed for a week.  Big thanks SBFSpot/PVOutput :-)
